src/Build/BackEnd/Components/RequestBuilder/TaskHost.cs

```diff
@@ -123,7 +126,10 @@ public TaskHost(IBuildComponentHost host, BuildRequestEntry requestEntry, Elemen
             _targetBuilderCallback = targetBuilderCallback;
             _continueOnError = false;
             _activeProxy = true;
-            _callbackMonitor = new Object();
+            _callbackMonitor = new object();
+            _disableInprocNode = ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_0)
```

Comment:
NIT: Could it break anyone? If I understand these changes correctly, new code can only make previously failing builds pass, doesn't it?

Reply:
I remember that back when I did this change I had some good reasons, but I forget what they were. Either way, for now it's protected under a change wave and whenever we remove Wave17_0, we'll remove the old code as well.